### ISSUE TITLE
Add data from excel sheet to database

### ISSUE THREAD

**Issue by prabhakarpd7284** [prabhakarpd7284/lab-booking-portal#2]
Currently admin has to add tool information one-by-one. The aim is to make this job easier. The admin should be able to update lab equipment information by just uploading an excel sheet containing all lab equipment information in proper format.

https://itsolutionstuff.com/post/laravel-56-import-export-to-excel-and-csv-exampleexample.html
http://www.expertphp.in/article/laravel-5-maatwebsite-import-excel-into-db-and-export-data-into-csv-and-excel
https://www.kodementor.com/laravel-excel-import-and-export-example-from-scratch/

**Comment by bhu800** [comment]
bhu800

**Comment by prabhakarpd7284** [comment]
To avoid ldap login, try this.
1. comment all ldap authentication code in app/Http/middleware/RedirectIfauthenticated.php
2. Add a username and password in db of ur choice.
3. Login using newly added username and password.
See if it works.

**Comment by bhu800** [comment]
sorry sir, was out of town for some time...
this is giving error...
![Screenshot (114)](https://user-images.githubusercontent.com/42897290/60769715-e967cd00-a0f0-11e9-9181-942c934617f4.png)

**Comment by prabhakarpd7284** [comment]
Can you upload app/Http/middleware/RedirectIfauthenticated.php file

**Comment by bhu800** [comment]
you asked to comment all ldap authentication code...
[RedirectIfAuthenticated.zip](https://github.com/prabhakarpd7284/lab-booking-portal/files/3365919/RedirectIfAuthenticated.zip)

**Comment by prabhakarpd7284** [comment]
You have to comment only ldap verification part in it.

**Comment by prabhakarpd7284** [comment]
Basically you comment from line 27-127 only in the file that you've sent.

**Comment by bhu800** [comment]
![Screenshot (115)](https://user-images.githubusercontent.com/42897290/60771846-35277000-a10b-11e9-8610-f55b380cff12.png)
on commenting 30-127 lines...

**Comment by prabhakarpd7284** [comment]
your mysql configuration is causing problems.

**Comment by prabhakarpd7284** [comment]
try updating **config/database.php** and **.env** files accordingly.

**Comment by bhu800** [comment]
error resolved but still it is redirecting to same page with being logged in

**Comment by bhu800** [comment]
problem solved...

**Comment by bhu800** [comment]
it was due to that i made user email and password from phpmyadmin while app was trying to match hashed password...
so i then registered from /register page and then used that user name and password...

**Comment by prabhakarpd7284** [comment]
good job. Now you can focus on the actual issue here. Keep going.

**Comment by bhu800** [comment]
sir tools import export method via excel sheet is now working properly...

**Comment by bhu800** [comment]
![Screenshot (116)](https://user-images.githubusercontent.com/42897290/60772259-5a1ee180-a111-11e9-832e-009d606aecfe.png)

**Comment by bhu800** [comment]
but there is a issue that it only download tools data in xlxs format ...
will try to do change it so that user have option to download it in xls, csv or xlxs format...

**Comment by bhu800** [comment]
also should I do it's styling also...

**Comment by bhu800** [comment]
also I wanted to ask how is someone assigned admin boolean in our system... as I made myself admin from phpmyadmin...
also it is showing option to add tools for both admin and normal user...

**Comment by prabhakarpd7284** [comment]
Only an admin should be able to make other user an admin. This functionality also needs to be added. I'll make a separate issue for that. Currently, its done by changing value in database.

Add tools should only be accessible by a user who has admin privileges. Try verifying this logic by logging by another user who is not an admin.

**Comment by bhu800** [comment]
yes for normal user(non-admin) it is redirecting to booking calender page....

**Comment by bhu800** [comment]
> but there is a issue that it only download tools data in xlxs format ...
> will try to do change it so that user have option to download it in xls, csv or xlxs format...

![Screenshot (117)](https://user-images.githubusercontent.com/42897290/60824708-b2181f80-a1c7-11e9-84d4-d5368cfbb837.png)
